packages/core/src/index.ts

```diff
@@ -233,8 +233,18 @@ export class Server<TContext extends InitialContext, TRootValue> {
 
     if (typeof options.graphiql === 'object' || options.graphiql === false) {
       this.graphiql = options.graphiql
-    } else {
+    } else if (options.graphiql === true) {
+      if (introspectionDisabled) {
+        // TODO: inform user that Graphiql may not work with introspection disabled
```

Comment:
Agree, should we just do `console.warn`?

Reply:
logger.warn is fine I guess.

Reply:
Added